CDLUC3/dmsp_frontend_prototype
# Built User Profile page with Email Authentication section

## Description

Built the User Profile page, which includes the ability to edit your user profile info, as well as delete emails, make an email your primary email, and add new email aliases.

1. Created a separate UpdateEmailAddress component
2. Created EmailAddressRow component
3. Created shared Form components, FormInput and FormSelect
4. Created temporary RightSidebar component
5. Updated TypeAheadWithOther component to work with parent components
6. Added some shared css for shared form components in a new style file called _form.scss
7. Added corresponding unit tests

I know that there is an updated, shared container and right sidebar template coming. I will create a separate ticket to update the Profile page to use that template structure after it has been merged.

Fixes # ([124](https://github.com/CDLUC3/dmsp_frontend_prototype/issues/124))

## Type of change

## How Has This Been Tested?
I tested this manually and by adding unit tests and running all unit tests for the app.

To test:
1) Start up dmsp_backend_prototype's development branch
2) Start up this branch of dmsp_frontend_prototype branch
3) Log in to the site to get the required tokens at http://localhost:3000/login
4) Go to http://localhost:3000/account/profile
5) Click on the "Edit" button of the "Your Profile" section
  -- Test updating the form fields and click "Update" button, to save your changes and be taken back to the "non-form field view". Check users db table to see that the updates are made accordingly.
  -- Leave one of the form fields blank and click "Update" to get field-level error messages from the client-side validation
  -- Delete your "dmspt" access token cookie. Click "Edit" button and then the "Update" button, and you should see your token refreshed. Repeat test of deleting access token and submitting Your Profile form, or selecting a new primary email, and you should see the token successfully refresh.

6) Email tests
  -- Add a new email alias at the bottom of the page, or delete an email from the list, or make a different email your primary, 
 and you should see the page re-renders with the new email added to the list. Check the db table, and you should see that the data updated in userEmails accordingly
7) Note Right sidebar and links. When you switch to a mobile device, you should see the right sidebar wrap

**NOTE:** The email service has not yet been implemented, so you will not be sent any emails to confirm the addition of new email aliases.

## Checklist:


## Screenshots
Top of page with "Your profile" section
<img width="1123" alt="image" src="https://github.com/user-attachments/assets/48b090b7-992b-4429-aa6d-8053e8643723">

Bottom of page with "Email and Authentication"
<img width="773" alt="image" src="https://github.com/user-attachments/assets/737a8bbe-45ca-47fd-a2a3-7b1600a60a44">
